### ISSUE TITLE
Child Windows not shown in FlaUInspect-1.0.0 but same is visible in VisualUIAVerify.exe 

### ISSUE THREAD

**Issue by kiransirigireddy** [Roemer/FlaUI#58]
Hi,

I am getting a pop up window in my desktop application which i am not able to get a handle using FlaUI framework.
If i inspect the same from FlaUInspect-1.0.0 also not able to see the window but same is visible from VisualUIAVerify.exe.

Tried by app.GetMainWindow(automation); and app.GetAllTopLevelWindows(automation); but still no luck. 
I am about to complete Automation POC on my 7 Desktop Enterprise Desktop applications and this is the only situation which is stopping to development on Smartbear Test Complete and jump on to FlaUI. 

Tried with Winium tool also before and got struck at the same point.
Please let me know if any more info required.

The application is developed in Progress.

Attached the screenshot of both FlaUIInspect and VisualUIAVerify after greying out the sensitive data.
The Window Object marked in yellow in VisualUIAVerify1.png is not present in FlaUInspect1.png.
![flauinspect1](https://cloud.githubusercontent.com/assets/27776775/26281854/22dc7232-3e21-11e7-96cd-82f8187f57f0.png)
![visualuiaverify_1](https://cloud.githubusercontent.com/assets/27776775/26281858/2b90716c-3e21-11e7-9198-863372d91fa5.png)

**Comment by Roemer** [comment]
Have you tried the "Hover Mode" in FlaUInspect?

**Comment by kiransirigireddy** [comment]
Yes Roemer. The screenshot i provided is HOver Mode only.it lands on the main window only.
Again attached latest screenshot.
![flauinspect2](https://cloud.githubusercontent.com/assets/27776775/26284222/da44dda6-3e53-11e7-8383-2225a03ef917.png)

**Comment by Roemer** [comment]
That is very strange. Is FlaUInspect started as admin (just to be sure).
Also could you try to use the treewalker to find the missing window? This can be done if you normally search for the "winPanel" window and then use the treewalker to get the childs child (see https://github.com/Roemer/FlaUI/wiki/TreeWalker and use the RawTreeWalker).

**Comment by kiransirigireddy** [comment]
Roemer,

Yes launched FlaUIInspect as Admin only and also in the meanwhile tried going by 
Window.FindChildren()[0].FindChildren()[].FindChildren()[0]

Able to see what is visible in FlaUIInspect only , Two Panes only below a Window but where as Visual UI Automation Verify shows a Window and Two Panes below window as shown in below Screenshot.

Please let me know if any more information is required.

![visualuiaverify_1](https://cloud.githubusercontent.com/assets/27776775/26294518/00629c6a-3ee2-11e7-9314-8d019579d663.png)

**Comment by Roemer** [comment]
Please try the RawTreeWalker as that one could see more than the normal "Find" methods.

**Comment by kiransirigireddy** [comment]
Thanks Roemer for your quick response, I don't have idea how i can do this but still let me try and get back to you.

**Comment by RafaelUbaldo** [comment]
Facing this exact issue right now, testing a very old Delphi app. The submenus are completely inaccessible. Key to note: they do no appear in Inspect until I first click the main menu (e.g. file), then I must most over to descend into sub menu items. How can I mock that behavior in FlaUI if the I can not click on MenuItems?

**Comment by Roemer** [comment]
That is strange. I've never seen that behavior. Is there any chance to get access to one of the apps? Maybe with just all the logic removed and just the ui remaining?

**Comment by Roemer** [comment]
@RafaelUbaldo That behavior is very normal for Win32 menus. FlaUI encapsulates this behavior for ContextMenus and also for menu if you have the latest code from github (fixed that yesterday, you need to set the IsWin32Menu on the menu element).

**Comment by JuniorHacks** [comment]
`// Top level menu
            var ApplicationMenuBar = COMMANDqc.FindFirstDescendant(ctrl => ctrl.ByName("Application")).AsMenu();
            Assert.That(ApplicationMenuBar, Is.Not.Null);

            var FilesMenu = ApplicationMenuBar.FindFirstChild(ctrl => ctrl.ByName("Files")).AsMenu();
            var FilesMenuItem = ApplicationMenuBar.MenuItems["Files"].AsMenuItem();
            Assert.That(FilesMenu, Is.Not.Null);
            Assert.That(FilesMenuItem, Is.Not.Null);
            Assert.That(FilesMenuItem.Equals(FilesMenu));
            
            // @todo: confirm first child theory.
            
            // Requery the Windows automation tree.to track promotion of files to Menu
            var RefreshedFilesMenu = COMMANDqc.FindFirstDescendant(ctrl => ctrl.ByName("Files")).AsMenu();
            Assert.That(RefreshedFilesMenu, Is.Not.Null);
            Assert.That(FilesMenu.Equals(RefreshedFilesMenu));
           
            var GlobalMenuItem = FilesMenu.MenuItems["Global"];
            Assert.That(GlobalMenuItem, Is.Null);
            var GlobalMenu = COMMANDqc.FindFirstDescendant(ctrl => ctrl.ByName("Global"))?.AsMenu();
            Assert.That(GlobalMenu, Is.Null);

            // When I focus on the Files Menu.
            FilesMenu.Focus();
            GlobalMenu = COMMANDqc.FindFirstDescendant(ctrl => ctrl.ByName("Global"))?.AsMenu();
            Assert.That(GlobalMenu, Is.Not.Null);

            // Focus vs mouse...
            //GlobalMenu.Focus();
            var ContactMenuItem = GlobalMenu.MenuItems["Contacts"];
            Assert.That(ContactMenuItem, Is.Null);
            
            // Let's the move the mouse over the Global Menu
            Mouse.Position = GlobalMenu.GetClickablePoint(); 
            // I neeed to pause to let the automation tree update
            Thread.Sleep(500);
            Assert.IsNotNull(GlobalMenu); // check lease

          
            // when a menu expands, it get's reattached as a new tree node (it's also the first child?)
            var TreeNodeInFocus = COMMANDqc.FindFirstChild();
            Assert.That(TreeNodeInFocus.Properties.Name,Is.EqualTo("Global")); // global node moved
            Assert.That(TreeNodeInFocus, Is.Not.EqualTo(GlobalMenu.AsMenu())); // different handler
            Assert.That(TreeNodeInFocus, Is.Not.Null);
            var ContactMenuItemUnderNewTree = TreeNodeInFocus?.AsMenuItem().SubMenuItems["Contacts"];
            Assert.That(ContactMenuItemUnderNewTree, 

> Is.Not.Null);

Here's the failure in context (maybe I am just doing it wrong as I have been coding UIA for under a week). In old Delphi apps, when menus are accessed, the submenu does not exist until the parent menu item has focus, at which time the new opened submenu options are attached to the automation tree somewhere else under the parent window as a top level menu. They appear to move around and detect depending on focused states. Not sure what I am looking at. I've seen the same behavior for my DevExpress grids. Along with elements not even visible in Inspect or accessible.

**Comment by Roemer** [comment]
@JuniorHacks Can you try the current version on master and use the Menu and SubMenuItems (set the IsWin32Menu on the Menu).

**Comment by kiransirigireddy** [comment]
Hi Roemer,

Tried the way you suggested Found the Pop up Window is still not visible with RawTreeWalker also.
Please find below screenshot with detailed Visual Studio Quick Watch Screenshots and VisualUIAVerify screenshots.

If i focus VisualUIAVerify tool and refresh the Parent of the Window even in VisualUIAVerify its is not displaying. Seems to be focus on the pop up is going off when i go to VisualUIAVerify 


Any Help on this will be usefull.I understand without showing the application it will be difficult to debugg.If you are ok for webex then i can check in my office if they are ok to share the screen of laptop.
![rawtreewalker](https://cloud.githubusercontent.com/assets/27776775/26351478/7c81d4de-3fd5-11e7-9a00-ce8c74a3f5b0.png)

**Comment by JuniorHacks** [comment]
@Roemer It's an old Win32 app. I have also noticed it behaves different in UIA2. I will investigate thoroughly and post back a write up.
